torch/distributed/_composable/fsdp/_fsdp_collectives.py

```diff
@@ -442,6 +442,8 @@ def _get_all_gather_input_metadatas(
             input_numels.append(all_gather_input.numel())
         param_all_gather_input_dtypes.append(input_dtypes)
         param_all_gather_input_numels.append(input_numels)
+    if all_gather_dtype is torch.float8_e4m3fn:
```

Comment:
Should this be for both float8 dtypes?

Reply:
good question! In total we have 4 float8 dtypes: https://github.com/pytorch/pytorch/blob/21ffa18ad1ad0bf22df8d7f42309190b4f3cc547/torch/storage.py#L524-L531

@vkuzo @drisspg could I assume parameters are always `float8_e4m3fn`? or should I list 4 float8 dtypes here?

@awgu this PR unblocks unit test with single Float8Linear module. Without the PR, we use `torch.uint8` only when dtypes are mixed. When passing float8 dtype to `foreach_copy_`, it throws error of `not being impelmented for float8 dtypes`

Reply:
Should we just support float8 dtypes in `_foreach_copy_`?

Reply:
So we need all 4 for AMD support, at least the 2 e4 variants. I think Andrew's suggestion is right

Reply:
> Should we just support float8 dtypes in `_foreach_copy_`?

@awgu do you mean improving existing cuda kernels `foreach_tensor_copy_list_kernel_cuda_` to cover float8? or improve at python level?

Reply:
the former
my current understanding is that this should only require changing the dispatch macro for what dtypes are supported

Reply:
thanks for the pointers! I can give it a try